nservicebus/azure-service-bus/configuration/configuration.md

```diff
@@ -0,0 +1,200 @@
+ * `BatchFlushInterval(TimeSpan)`: This setting controls the batching behavior for message senders. They will buffer send operations during this time frame and send all messages at once. Defaults to 0.5 seconds. Specify TimeSpan.Zero to turn batching off.
+ * `MessagingFactorySettingsFactory(Func<string, MessagingFactorySettings>)`: This factory method allows to override creation of messaging factories.
+ 
+### Message Receivers
+
+The following settings determine how NServiceBus will receive messages from Azure Service Bus entities.
+
+ * `ReceiveMode(ReceiveMode)`: Determines part of the 'transactional behavior'. `ReceiveMode.PeekLock` ensure that messages will be only removed from the queue after processing is done. If processing fails, or takes to long, the message will reappear on the queue for re-processing. Basically emulating transactional rollback. `ReceiveMode.ReceiveAndDelete` will delete the message immediately after receive, meaning there is no rollback when an exception occurs. Defaults to `ReceiveMode.PeekLock`.
+ * `PrefetchCount(int)`: This setting will make the receiver prefetch messages on receive operations, and acts like client side batching. Defaults to 200, meaning that each receive operation will pull in another 200 messages.
+ * `RetryPolicy(RetryPolicy)`: Determines how the receiver should respond on transient connectivity failures. Defaults to `RetryPolicy.Default`, which is an exponential retry under the hood.
+ * `AutoRenewTimeout(TimeSpan)`: When using `ReceiveMode.PeekLock`, the broker will lock a received message for a period specified as `LockDuration(TimeSpan)` on the entity. If processing doesn't end within this time period, a message will reappear on the entity and will be reprocessed. For long running operations this might imply that the operation will be executed multiple times in parallel, which is usually undesirable. This setting allows to extend the lock automatically when processing hasn't finished yet. Ensure that the `TimeSpan` specified here is a little smaller than the `TimeSpan` specified as `LockDuration`.
+
+### Message Senders
+
+The following settings determine how NServiceBus will send messages to Azure Service Bus entities.
+
+ * `RetryPolicy(RetryPolicy)`: Determines how the sender should respond on transient connectivity failures. Defaults to `RetryPolicy.Default`, which is an exponential retry under the hood.
+ * `BackOffTimeOnThrottle(TimeSpan)`: 'As A Service' products will tell there clients to back off under heavy load and so will Azure Service Bus. This setting specifies how long NServiceBus will wait to try sending again in these occurances. Defaults to 10 seconds. Note, by default this backoff period will increase exponentially during retries. 
+ * `RetryAttemptsOnThrottle(int)`: Defines how many times the transport should attempt to retry a send operation when being operated before throwing an exception.
```

Comment:
yes, services are remote and thus subject to a lot of potential network problems + throttling, all may lead to potential transient failures.

No it's not the same, the SDK will just give up sending and throw. It's up to the client code to handle that exception. If in context of an nsb handler this will result in the received message becoming visible again and therefore a retry. If outside the context of an nsb handler, than the message may be lost unless the developer takes care of retrying some way.

How can I learn about this exception: Very good question, we don't know. There is a tracking ID in the exception message, which could be used by the ASB team itself to go figure out what happened.